bitmovin-analytics-datasource/src/components/FilterRow.tsx

```diff
@@ -0,0 +1,163 @@
+import React, { useState } from 'react';
+import { difference, isEmpty } from 'lodash';
+import { SelectableValue } from '@grafana/data';
+import { Box, HorizontalGroup, IconButton, InlineLabel, VerticalGroup } from '@grafana/ui';
+
+import { QueryFilter, QueryFilterOperator, QueryFilterValue } from '../types/queryFilter';
+import { QueryAdAttribute, SELECTABLE_QUERY_AD_ATTRIBUTES } from '../types/queryAdAttributes';
+import { QueryAttribute, SELECTABLE_QUERY_ATTRIBUTES } from '../types/queryAttributes';
+import { FilterInput } from './FilterInput';
+import { convertFilterValueToProperType } from '../utils/filterUtils';
+
+type Props = {
+  readonly isAdAnalytics: boolean;
+  readonly onChange: (newFilters: QueryFilter[]) => void;
+};
+
+type Filter = {
+  selectedAttribute: SelectableValue<QueryAdAttribute | QueryAttribute>;
+  selectedOperator: SelectableValue<QueryFilterOperator>;
+  rawFilterValue: string;
+  convertedFilterValue: QueryFilterValue;
+  parsingValueError: string;
+};
+
+export function FilterRow(props: Props) {
+  const [filters, setFilters] = useState<Filter[]>([]);
+
+  const mapFilterAttributesToSelectableValue = (): Array<SelectableValue<QueryAttribute | QueryAdAttribute>> => {
+    const selectedAttributes = filters.map((filter) => filter.selectedAttribute);
+    if (props.isAdAnalytics) {
+      return difference(SELECTABLE_QUERY_AD_ATTRIBUTES, selectedAttributes);
+    } else {
+      return difference(SELECTABLE_QUERY_ATTRIBUTES, selectedAttributes);
+    }
+  };
+
+  const mapFiltersToQueryFilters = (filters: Filter[]): QueryFilter[] => {
+    return filters.map((filter) => {
+      return {
+        name: filter.selectedAttribute.value!,
+        operator: filter.selectedOperator.value!,
+        value: filter.convertedFilterValue,
+      } as QueryFilter;
+    });
+  };
+
+  const addFilterInput = () => {
+    setFilters((prevState) => [
+      ...prevState,
+      {
+        selectedAttribute: {},
+        selectedOperator: {},
+        rawFilterValue: '',
+        convertedFilterValue: '',
+        parsingValueError: '',
+      } as Filter,
+    ]);
+  };
+
+  const onAddFilter = (index: number) => {
+    const filter = filters[index];
+    try {
+      const convertedValue = convertFilterValueToProperType(
+        filter.rawFilterValue,
+        filter.selectedAttribute.value!,
+        filter.selectedAttribute.label!,
+        filter.selectedOperator.value!,
+        props.isAdAnalytics
+      );
+
+      const newFilter = { ...filter, convertedFilterValue: convertedValue, parsingValueError: '' } as Filter;
+
+      const newFilters = [...filters];
+      newFilters.splice(index, 1, newFilter);
+
+      setFilters(newFilters);
+
+      props.onChange(mapFiltersToQueryFilters(newFilters));
+    } catch (e: any) {
```

Comment:
like `new Error('Oh no!')` type

```suggestion
    } catch (e: Error) {
```